### PR TITLE
From curves file method

### PR DESCRIPTION
**Pull request by joaobernardo23**

### CODE REVIEW

File: inductiva/stellarators/scenarios/coils/coils.py [hunk 1]
```diff
@@ -141,6 +141,59 @@ def from_random_curves(cls,
...
+                         num_coils,
+                         coil_currents,
+                         curves_file,
+                         delimiter=',',
+                         max_order=14):
+        """Create StellaratorCoils from Fourier coefficients loaded from a file.
+
+        This function loads a file containing Fourier coefficients for several 
+        coils. The file is expected to have `6*num_coils` many columns, and 
+        `order+1` many rows. The columns are in the following order,
```
Comment (fabiocruz) [inline inductiva/stellarators/scenarios/coils/coils.py:156]: So the parameters `num_coils` and `max_order` aren't needed, right? They can be inferred from the file.

File: inductiva/stellarators/scenarios/coils/coils.py [hunk 2]
```diff
@@ -141,6 +141,59 @@ def from_random_curves(cls,
...
+
+        sin_x_coil1,cos_x_coil1,sin_y_coil1,...,sin_x_coil2,cos_x_coil2,...
```
Comment (fabiocruz) [inline inductiva/stellarators/scenarios/coils/coils.py:158]: Use a consistent notation. You have previously called this s_j, sj, etc. Make it consistent everywhere you mention them in this class.

File: inductiva/stellarators/scenarios/coils/coils.py [hunk 3]
```diff
@@ -141,6 +141,59 @@ def from_random_curves(cls,
...
+                    curve_coefficients[:, order] = [
+                        float(val) for val in line[6 * curve:6 * (curve + 1)]
+                    ]
+
+                # Create the Coil object
+                coil = Coil(curve_coefficients, coil_currents[curve])
+                coils.append(coil)
+
+                # Reset the file pointer to the beginning
+                file.seek(0)
```
Comment (fabiocruz) [inline inductiva/stellarators/scenarios/coils/coils.py:193]: Can we simplify this considerable by using e.g. [this method](https://numpy.org/doc/stable/reference/generated/numpy.fromfile.html) to read an array from a text file?

**Review by fabiocruz — COMMENTED** [review]
First round of comments, aimed at readability and simplifying the code.

File: inductiva/stellarators/scenarios/coils/coils.py [hunk 4]
```diff
@@ -141,6 +141,51 @@ def from_random_curves(cls,
...
+            curves_file (str): Name of the file containing Fourier coefficients.
+            delimiter (str): Delimiter used in the file. 
+
+        Returns:
+            StellaratorCoils: The created StellaratorCoils instance.
+        """
+
+        # Reads all the coefficients from the file to a numpy 2D array
+        # with `order+1` rows and `6*num_coils` columns.
+        curves_data = np.loadtxt(fname=curves_file, delimiter=delimiter)
```
Comment (fabiocruz) [inline inductiva/stellarators/scenarios/coils/coils.py:170]: Much simpler!

**Review by fabiocruz — APPROVED** [review]
Thanks @joaobernardo23, this is much better!

File: inductiva/stellarators/scenarios/coils/coils.py (hunk 4 above)
Comment (joaobernardo23) [reply inductiva/stellarators/scenarios/coils/coils.py:170]: Definitely!

**Pull request merged**